### PR TITLE
FreeBSD/ports: First version of ports building files

### PR DESCRIPTION
**Pull request by wjwithagen**
Signed-off-by: Willem Jan Withagen <wjw@digiware.nl>

### CODE REVIEW

**Comment by wjwithagen** [comment]
@liewegas 
How would you suggest to get reviewers approval for getting this comitted?
There anyways a catch-22 in that the FreeBSD port will require a sha256 on the source, which 
then cannot be in the port-tree here in ceph, because adding the sha256 will change git and the actual sha256. ;-)

**Comment by liewegas** [comment]
@wjwithagen I'm not familiar with how the ports are managed so I can't really suggest anything there.  I'm happy to pull this into ceph.git, though, if it makes things easier; we do that for debian/ and ceph.spec as well.

File: FreeBSD/usr/ports/net/ceph-devel/distinfo [hunk 1]
```diff
@@ -0,0 +1,35 @@
+TIMESTAMP = 1486914967
```
Comment (tchaikov) [inline FreeBSD/usr/ports/net/ceph-devel/distinfo]: @wjwithagen i don't think we should have the port files in upstream repo as it includes a point-in-time meta info like `distinfo`. you can have this in your own dedicated repo for maintaining the port before it's accepted[1] by FreeBSD ports. and it's not advisable to keep the packaging in upstream unless we are able to test it using our test infrastructure.

---
[1] https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=217046

Comment (wjwithagen) [inline FreeBSD/usr/ports/net/ceph-devel/distinfo]: @tchaikov 
I sort of agree.
Certainly after I have had this rather lengthy review process for the port submision on:
    https://reviews.freebsd.org/D9584
They are not make my life any easier, and having to keep it upstreamed into Ceph would be more "service" to others in ceph as to how to build and package it.
But I'm not closing it for a bit of time to come, since I'm not certain that the port will be accepted in it's current state. Main pain point is /etc/ceph.
Ports are not allowed to modify /etc and/or store data in /etc.

So I'll have to rework configs/authkey stuff to be able to work with /usr/local/etc/ceph.
And then also import that knowledge into all kinds of scripts.
And get that accepted here, which will take quite some time.

So bottom line: lets keep this submission for the time being. It is a mark in history.
And once FreeBSD ports accepts the port, I'll close this PR.

**Comment by wjwithagen** [comment]
@tchaikov 
I do prefer to actually have it here as well, since FreeBSD-port only keeps a rather limited set of port/packages online. And I thing it is easier to start digging here for version, because that will also have all the other correct files in the same checkout.
